image-builder/builder-services/src/main/java/io/hyscale/builder/cleanup/services/impl/DeleteAfterBuild.java

```diff
@@ -35,21 +35,21 @@
 	private ImageCommandProvider imageCommandProvider;
 	
 	@Autowired
-	private DockerImageUtil dockerImageUtil;
+
+	private ImageUtil imageUtil;
 
 	@Override
 	public void clean(ServiceSpec serviceSpec) {
-		String image=dockerImageUtil.getImageName(serviceSpec);
-		List<String> imageIds = Arrays
-				.asList(CommandExecutor.executeAndGetResults(imageCommandProvider.getAllImageCommandByImageName(image))
-						.getCommandOutput().split("\\s+"));
-		
-		CommandExecutor.execute(imageCommandProvider.getAllImageDeleteCommand(new HashSet<String>(imageIds)));
+		String image=null;
+		try {
+			image = imageUtil.getImage(serviceSpec);
+		} catch (HyscaleException e) {
+			e.printStackTrace();
```

Comment:
No printing stack trace ...better log the exception and warn user about it through WorkflowLogger